tests/handlers/aws/test_handler.py

```diff
@@ -545,8 +546,387 @@ def test_lambda_handler_failure(self) -> None:
                 handler(event, ctx)  # type:ignore
 
 
+def _wait_for_container(container: Container, port: str) -> None:
+    while port not in container.ports or len(container.ports[port]) == 0 or "HostPort" not in container.ports[port][0]:
+        container.reload()
+        time.sleep(1)
+
+
+def _wait_for_localstack_service(wait_callback: Callable[[], None]) -> None:
+    while True:
+        try:
+            wait_callback()
+        except AssertionError:
+            time.sleep(1)
+        else:
+            break
+
+
+def _create_secrets(secret_name: str, secret_data: dict[str, str], localstack_host_port: str) -> Any:
+    client = aws_stack.connect_to_service(
+        "secretsmanager", region_name="eu-central-1", endpoint_url=f"http://localhost:{localstack_host_port}"
+    )
+    client.create_secret(Name=secret_name, SecretString=json.dumps(secret_data))
+
+    return client.describe_secret(SecretId=secret_name)["ARN"]
+
+
+def _upload_content_to_bucket(content: Union[bytes, str], content_type: str, bucket_name: str, key_name: str) -> None:
+    client = aws_stack.connect_to_service("s3")
+
+    client.create_bucket(Bucket=bucket_name, ACL="public-read-write")
+    client.put_object(Bucket=bucket_name, Key=key_name, Body=content, ContentType=content_type)
+
+
 @pytest.mark.integration
-class TestLambdaHandlerSuccess(TestCase):
+class TestLambdaHandlerSuccessKinesisDataStream(TestCase):
+    def setUp(self) -> None:
+        docker_client = docker.from_env()
```

Comment:
Just wanted to highlight [`testcontainers-python`](https://testcontainers-python.readthedocs.io/en/latest/index.html). Have you considered using it instead of plain docker? Could it help reducing boilerplate?

Reply:
nice! I didn't know about that.
I have to investigate the feature of the provided `LocalStack` container.
the boilerplate is mostly about getting the random port opened by docker on the host and override it for the `localstack` utils package.

I will open an issue in order to check how much is feasible

Reply:
https://github.com/elastic/elastic-serverless-forwarder/issues/67